### ISSUE TITLE
Next beta/feature release

### ISSUE THREAD

**Issue by Lineflyer** [cgeo/cgeo#12451]
It should be the time now to think about bringing current `master` (including bottom navigation) into a beta state in preparation of a new feature release (unless there is some critical regression or website change).

This also means we should stop adding new (untested) features at some point to be able update `release` to the head of `master` and have a more-or-less stable version. 

As the functions around waypoint calculation and variable tab are still developed in parallel to existing functions and (by default) hidden from end users its uncritical to continue work on this. Or shall this be completed first to be included unhidden in the next beta?

I suggest to check the open feature-PR whether to be included before or after branching and provide feedback here.
Same goes for issues: Do we have to-be-fixed regressions or problems in new features needing to be fixed before branching or before releasing a beta?

My first quick look (feel free to edit and append):
- [x] #12448 <-- Should be merged to mitigate #9992
- [x] #12441 <-- Include into next beta?
- [x] #12443 <-- Should be decided upon and fixed before beta
- [x] #12423 <-- Should be decided upon and fixed before beta
- [x] #12425 <-- Regression, which needs to be fixed before beta
- [x] #12409 <-- Deadline 04/2022
- [x] #12457 <-- Regression, which needs to be fixed before beta
- [x] #12558 
- [x] #12559
- [x] #12557
- [x] #12613
- [x] #11969 (remaining tasks)
- [x] #12654
- [x] #12667 Missing log count (which also causes missing fav percentage)
- [x] #12673 Log upload with pictures
- [x] #12659 Upcoming field note page change
- [x] #12452
- [x] #12689
- [x] #12688
- [x] #12701 (for release add temporary mitigation at least: add "clear manual targets" to menu)
- [x] #12704
- [x] #12710 <-- Regression, which needs to be fixed before release
- [x] #12573

**Comment by eddiemuc** [comment]
Wrt waypoint calculation: I wouldn't recommend waiting for completion before next beta. There's still much work to do befor this can be presented to (beta) users and features might still break. Whole PN export/import and migration is still open, and I am not satisfied with solution ideas for the "guides input" yet at all.

**Comment by eddiemuc** [comment]
Hm oraBe active in debug mode?

**Comment by ztNFny** [comment]
I'd add https://github.com/orgs/cgeo/teams/team/discussions/52 to that list

**Comment by moving-bits** [comment]
Added #12462 to fix #12457.

What about #12418? Should be non-critical as it adds a single menu entry (for OSM offline maps only), displaying an info dialog.

**Comment by Lineflyer** [comment]
> What about #12418? Should be non-critical as it adds a single menu entry (for OSM offline maps only), displaying an info dialog.

IMHO if its ready we can add it, but we do not need to wait for it.

**Comment by Lineflyer** [comment]
Regarding next beta planning we agreed in todays dev call on following:
- Current feature state on `master` looks OK for beta
- Bottom nav "show or not show" issues (#12423, #12443) shall be implemented as agreed
- Beta should not wait for the features "Multiple track files" and "New variable tab + new WP calc"

**Comment by Lineflyer** [comment]
Looking at the above list, it seems we are almost where we wanted to be.

What is the status of the security issue #12409?

I suggest to throttle down activity on master (no merging of additional new features only fixing of bugs, etc.) and look into the nightly performance and stability next days. Then decide on branching for beta.

**Comment by moving-bits** [comment]
I propose to include #12525 (add setting to limit distance on nearby search), as this is frequently requested on support.

#12507 (multiple track files) is ready to merge, can be either included or postponed, probably depending on how long next beta phase is planned to be.

**Comment by Lineflyer** [comment]
> #12507 (multiple track files) is ready to merge, can be either included or postponed, probably depending on how long next beta phase is planned to be.

I guess we will for the upcoming beta have a rather long phase (unless needed to release for some website change). Thus we could include it IMHO if you are "on standby" for fixing regressions.

> I propose to include #12525 (add setting to limit distance on nearby search), as this is frequently requested on support.

Yes, that sounds reasonable.
Did you see my comment whether to limit it to nearby or to all coord related searches?

**Comment by moving-bits** [comment]
> > #12507 (multiple track files) is ready to merge, can be either included or postponed, probably depending on how long next beta phase is planned to be.
> 
> I guess we will for the upcoming beta have a rather long phase (unless needed to release for some website change). Thus we could include it IMHO if you are "on standby" for fixing regressions.

I am (as time permits).

> > I propose to include #12525 (add setting to limit distance on nearby search), as this is frequently requested on support.
> 
> Did you see my comment whether to limit it to nearby or to all coord related searches?

I had it that way first, applying it to all three such searches, but changed it on @eddimuc's request. 
At least for "live mode" on the map it should not be applied.
For address and coordinate search it probably depends on your use case. If I think of "planning for next geocaching trip during holiday", it probably should not be applied either, whereas if you are using those two as an alternative way for nearby search (just at a different location), it might get applied. But how to distinguish between those use-cases? I tend to apply it to the explicit "nearby" search only.

**Comment by Lineflyer** [comment]
> I tend to apply it to the explicit "nearby" search only.

Yeah, lets start from there and listen to users feedback.

**Comment by Lineflyer** [comment]
Regarding #12409

I just updated the internal thread about that one.
If we trust the latest Google Play report, than that issue is no longer existing on `master` (tested with 2022.01.15-NB).

So IMHO we can go for a beta. If it shows up again we can still check what to do.

Any ongoing implementations or would we read to merge branches?

**Comment by eddiemuc** [comment]
> Any ongoing implementations or would we read to merge branches?

Nothing from my side since Vars are not relevant for the beta

**Comment by moving-bits** [comment]
I've just merged the "multiple tracks", as discussed above. Currently nothing else in the pipeline from my side.

We should include #12540 before branching. (Have you seen my comment?)

**Comment by Lineflyer** [comment]
> We should include #12540 before branching. (Have you seen my comment?)

Yes, and updated accordingly. Requested your review again. Feel free to merge.

**Comment by Lineflyer** [comment]
#12540 is merged. Will fast-forward `release` soon in preparation of next beta. 
Be a little patient before merging other things to `master`.

**Comment by Lineflyer** [comment]
Regarding #12409

I just updated the internal thread about that once more.
The Play report for the next nightly shows both warning again...seems unreliable.

Any idea about that one?

I will anyway continue branching somewhen today and keep you informed.

**Comment by Lineflyer** [comment]
Please note:

I just merged `master` into `release` in preparation of next beta.
This means from now on, please target all changes for next beta (if any) to `release` branch.
`master` is free to be used for new stuff.

@moving-bits 
I followed your Wiki guidance for changelogs (i.e. empty changelog_bugfix before merging).
However: Isn't there the need to empty changelog_base on `master` after the merge?

**Comment by fm-sys** [comment]
I've just opened an issue about visual inconsistencies in the new "multi trackfile"-popup, which should IMHO be fixed before/during our beta:  #12552

**Comment by fm-sys** [comment]
> I've just opened an issue about visual inconsistencies in the new "multi trackfile"-popup, which should IMHO be fixed before/during our beta: #12552

I've created #12558 for it...

**Comment by fm-sys** [comment]
Oh, and also consider #12559 before releasing ;-)

**Comment by Lineflyer** [comment]
added those above.

As @moving-bits is AFK this week we might wait for him to take a look into that PRs though.
Beta is not urgent anyway as we already branched.

**Comment by Lineflyer** [comment]
I marked #12409 as done here assuming the PR by @eddiemuc solved it.

**Comment by moving-bits** [comment]
> @moving-bits I followed your Wiki guidance for changelogs (i.e. empty changelog_bugfix before merging). However: Isn't there the need to empty changelog_base on `master` after the merge?

IIRC clearing the `changelog_main.md` happens directly before releasing the feature version. By this you can still re-do branching if need arises, as all contents of the yet-unreleased version are still available on `master`.

Also, all changes on `release` before doing the feature release go into `changelog_main.md` as well.

With releasing the feature release, `changelog_main.md` gets emptied on `master`, and stays unchanged on `release`. All further changes on `release` go into `changelog_bugfix.md` then.

**Comment by Lineflyer** [comment]
I added #12557 as blocking point for beta. It seems to me, that routing is not functional at all on GMaps.

**Comment by Lineflyer** [comment]
#12558 has been merged (discussion about final design might continue in the related issue though).

I am preparing a beta version now.

**Comment by Lineflyer** [comment]
Waiting with beta release to let you check #12577

**Comment by Lineflyer** [comment]
As CI failed working yesterday and I was unsure about #12579 I suggest to push the beta after the weekend.
Maybe also @mucek4 finds time to reset those failing executors until then.

**Comment by Lineflyer** [comment]
CI is back working. I will now prepare the beta version if no one stops me.

**Comment by Lineflyer** [comment]
Beta version 2022.01.22-RC is rolling.

**Comment by Lineflyer** [comment]
As we do have some fixes on `release` (especially also for #12616 which bothers some users) I would like to update the beta version as soon as possible.

Any ongoing fixes before doing that?
What about:
- [ ] #12613
- [ ] #11969 (remaining tasks)

**Comment by eddiemuc** [comment]
Side question regarding the current beta: I expected a lot of feedback for the new bottom navigation since it is a visually very prominent change. Is it really very quiet around this topic or am I overlooking something?

**Comment by Lineflyer** [comment]
> Side question regarding the current beta: I expected a lot of feedback for the new bottom navigation since it is a visually very prominent change. Is it really very quiet around this topic or am I overlooking something?

Yes, indeed very quiet, which I think is very positive, only:
- Some few beta users with comments like "The old was better". Seems the cache icon change was more "invasive" to users.
- Some beta users missing the "Go to" (I prepared a FAQ entry for that one)
- Some beta users with missing/wrong avatar (already addressed)

But might also be, that due to the current season, the usage is lower than usual.
IMHO we should therefore keep the beta running for a few more weeks.

**Comment by ztNFny** [comment]
Well for the icons we had several posts actively asking for input. Outside of those we didn't get too much feedback either.
So if feedback is wanted i think we need to ask for it.

**Comment by fm-sys** [comment]
If something stopped working or the overall experience got worse, users would have already complained. So let's take it as a compliment for our good work ;-) 🎉

**Comment by ztNFny** [comment]
as we have a time limit on https://github.com/cgeo/cgeo/issues/12659 this should be included

**Comment by Lineflyer** [comment]
Not sure why this was "unpinned", pinned it again ;)

As mentioned by @ztNFny we have three regressions based on website changes:
- [x] #12667 Missing log count (which also causes missing fav percentage)
- [ ] #12673 Log upload with pictures
- [ ] #12659 Upcoming field note page change

It might be wise to release a new version asap (or latest before February, 15th due to #12659).
While its technically possible to publish a release base don current production + fixes, I would prefer to release based on the current beta, as:
- Beta seems to perform rather goog
- Beta users would otherwise be thrown back (at least temporarily) to the non-bottom nav version

**Comment by Lineflyer** [comment]
FAQ entries added for the time being:
https://www.cgeo.org/faq#12667
https://www.cgeo.org/faq#12673

**Comment by ztNFny** [comment]
> Beta users would otherwise be thrown back (at least temporarily) to the non-bottom nav version

That could be avoided by releasing a new beta at the same time

**Comment by moving-bits** [comment]
I've added #12452 and consolidated all issues into the first article of this issue

**Comment by Lineflyer** [comment]
Two PRs outstanding and maybe some testing after merged (upstream) before another (last) beta round?!

**Comment by Lineflyer** [comment]
Time is ticking.
Do we need to wait for #12701 and #12704 which have been inserted into the initial post?

**Comment by moving-bits** [comment]
> Time is ticking. Do we need to wait for #12701 and #12704 which have been inserted into the initial post?

#12704 is implemented already, can be merged IMHO (see #12707).

I'm working on #12701, but it's a bit tricky, not sure if I can finish it today. Alternatively I can prepare a PR as mitigation, which simply adds the "Clear manual targets" to the menu.

**Comment by ztNFny** [comment]
What about  Fieldnotes upload behaves weird #12702 ?

**Comment by Lineflyer** [comment]
> What about Fieldnotes upload behaves weird https://github.com/cgeo/cgeo/issues/12702 ?

That IMHO can be done after release and with collected feedback by users. Most of that issue is to be solved on server side. I would only strive for a mitigation on our side once the situation is claer (maybe Groundspeak also optimized that faulty thing).

> Have implemented a shot at https://github.com/cgeo/cgeo/issues/12701. It's a bit tricky, so a review / some testing would be welcomed, see PR https://github.com/cgeo/cgeo/pull/12709.

How important is this manual target removal?
IMHO we could use the time until another bugfix release for checking/testing. Until now no beta user complained about that (and I even didn't know it existed).

**Comment by moving-bits** [comment]
> How important is this manual target removal?

If we do not want to go live with the "button magic" now (PR #12709) so shortly after implementing it (which I would understand), we should at least implement the mitigation of re-adding the menu entry "Clear manul targets", as the current way of resetting manual targets is pretty hidden (as described in #12701).

**Comment by fm-sys** [comment]
We have another IMHO important regression:   #12710

**Comment by fm-sys** [comment]
I've just read through the feedback on Facebook, sounds surprisingly positive. Seems like we've done a good job :)

**Comment by moving-bits** [comment]
I would vote for merging #12709 and then do either a short beta phase over the weekend (vote 👍), to be able to release in time for the website changes, or release directly (vote 🚀).

**Comment by ztNFny** [comment]
> All ToDo have been merged, so we can also test it ourselves in the nightlies for 1-2 days and release right after the weekend.

Why not a beta and use those 1-2 days with more users?

**Comment by Lineflyer** [comment]
We can try, but I doubt a lot of feedback:
- Google Play needs up to 24h to make the beta available
- Then users need to see the update and install it (maybe most will not do it right away)
- Crash reports arrive with a delay of 24-48 hours.

In other words:
If 15.02. is the deadline we plan to fulfill, then latest on 14.02. (better 13.02.) the release needs to be pushed to Google Play.
This will mean we would only have a 1 day beta...

**Comment by moving-bits** [comment]
_If_ we want to push another beta, we could do so tonight. After the weekend would be too late, IMHO.

**Comment by moving-bits** [comment]
> Edit: I think I will rather intensively use the nightly next days (and also look at the fixed areas) and then carefully prepare the release instead of rushing a beta.

fine for me as well

**Comment by moving-bits** [comment]
I have added #12573 to the list (and created PR #12715 for it) - we should do the renaming before releasing to avoid irritations when changing the label at a later point in time.

**Comment by Lineflyer** [comment]
All done here, will try some of the fixed things in nightlies over the weekend and encourage others to do alike.

**Comment by Lineflyer** [comment]
Triggered a new nightly build just now. So 2022.02.12-NB1 should be available in some minutes.

**Comment by Lineflyer** [comment]
We decided to release. Working on it:

- [ ] Check/Insert correct version name in ```/main/src/cgeo/geocaching/utils/BranchDetectionHelper.java``` (Format `YYYY-MM-DD`)
- [ ] Build release on [CI server](https://ci.cgeo.org/view/Release_Nightly/job/cgeo%20release/) (Download available after build from [here](https://download.cgeo.org/cgeo-release.apk))
- [ ] Perform smoke test with resulting ```cgeo-release.apk``` before continuing
- [ ] Upload and publish ```cgeo-release.apk``` to Google Play (Account required for the c:geo team Google Play account)
- [ ] Tag the release on Github in format `market_yyyymmdd` (can be combined with next step if done on Github UI)
- [ ] Publish the release on Github including upload of ```cgeo-release.apk``` to Github (Select Draft a new release from [here](https://github.com/cgeo/cgeo/releases), enter tag name (see above), make sure to select branch `release`, C&P change log, attach APK file, release the version)
- [ ] Publish release on F-Droid by runnning [this job](https://ci.cgeo.org/view/FDroid/job/Publish%20FDroid%20-%20RELEASE/) This required, that the release APK has been uploaded to Github in previous step
- [ ] Trigger notification to status.cgeo.org via [CI server](https://ci.cgeo.org/view/Notification%20system/job/Create%20release%20notification/) once the release is available on Google Play (not just being processed, but being available for download - sometimes takes up to one day)
- [ ] Post release info to Facebook 
- [ ] Post release info to Twitter

**Comment by Lineflyer** [comment]
Release 2022.02.13 is rolling. Will follow up tomorrow once its available on Google Play.

Thanks @cgeo/team for all your work and the excellent new features!

**Comment by Lineflyer** [comment]
That was fast.
Release is shown as live by Google Play. All steps are finished.

We are done!!

Note for watchers:
The Google Play app sometimes buffers version states, so it might still show the old release as up to date. Normally a reboot or force restart of the Google Play app helps.
